tests/test/org/commcare/cases/util/test/CasePurgeRegressions.java

```diff
@@ -35,4 +51,230 @@ public void testSimpleExtensions() throws Exception {
         }
     }
+
+    /**
+     * Test correct validation of a graph where 1 case indexes a non-existent node
+     */
+    @Test
+    public void testValidateCaseGraphBeforePurge_simple() throws Exception {
+        MockUserDataSandbox sandbox = MockDataUtils.getStaticStorage();
+        ParseUtils.parseIntoSandbox(this.getClass().getClassLoader().
+                getResourceAsStream("case_purge/validate_case_graph_test_simple.xml"), sandbox);
+        IStorageUtilityIndexed<Case> storage = sandbox.getCaseStorage();
+
+        HashMap<String, Integer> caseIdsToRecordIds = createCaseIdsMap(storage);
+        CasePurgeFilter filter = new CasePurgeFilter(storage);
+
+        Set<String> nodesExpectedToBeLeft = new HashSet<>();
+        nodesExpectedToBeLeft.add("case_one");
+        nodesExpectedToBeLeft.add("case_two");
+
+        Set<String[]> edgesExpectedToBeLeft = new HashSet<>();
+        edgesExpectedToBeLeft.add(new String[]{"case_two", "case_one"});
+
+        // Check that the edges and nodes still present in the graph are as expected
+        DAG<String, int[], String> internalCaseGraph = filter.getInternalCaseGraph();
+        checkProperNodesPresent(nodesExpectedToBeLeft, internalCaseGraph);
+        checkProperEdgesPresent(edgesExpectedToBeLeft, internalCaseGraph);
+
+        // Check that the correct cases were actually purged
+        Vector<Integer> expectedToRemove = new Vector<>();
+        expectedToRemove.add(caseIdsToRecordIds.get("case_three"));
+        Vector<Integer> removed = storage.removeAll(filter);
+        checkProperCasesRemoved(expectedToRemove, removed);
+    }
+
+    /**
+     * Test correct validation of a graph where 2 different cases index the same non-existent node,
+     * and both of those cases have child nodes
+     */
+    @Test
+    public void testValidateCaseGraphBeforePurge_complex() throws Exception {
+        MockUserDataSandbox sandbox = MockDataUtils.getStaticStorage();
+        ParseUtils.parseIntoSandbox(this.getClass().getClassLoader().
+                getResourceAsStream("case_purge/validate_case_graph_test_complex.xml"), sandbox);
+        IStorageUtilityIndexed<Case> storage = sandbox.getCaseStorage();
+
+        HashMap<String, Integer> caseIdsToRecordIds = createCaseIdsMap(storage);
+        CasePurgeFilter filter = new CasePurgeFilter(storage);
+
+        Set<String> nodesExpectedToBeLeft = new HashSet<>();
+        nodesExpectedToBeLeft.add("case_one");
+        nodesExpectedToBeLeft.add("case_two");
+
+        Set<String[]> edgesExpectedToBeLeft = new HashSet<>();
+        edgesExpectedToBeLeft.add(new String[]{"case_two", "case_one"});
+
+        // Check that the edges and nodes still present in the graph are as expected
+        DAG<String, int[], String> internalCaseGraph = filter.getInternalCaseGraph();
+        checkProperNodesPresent(nodesExpectedToBeLeft, internalCaseGraph);
+        //checkProperEdgesPresent(edgesExpectedToBeLeft, internalCaseGraph);
```

Comment:
commented out line should be removed, and also `edgesExpectedToBeLeft` is never read from (because of commented out code)